js/ui/payg.js

```diff
@@ -191,7 +195,13 @@ var PaygUnlockUi = GObject.registerClass({
     }
 
     validateCurrentCode() {
-        return Main.paygManager.validateCode(this.entryCode);
+        if (Main.paygManager.validateCode(Main.paygManager.codeFormatPrefix
```

Comment:
I think you should move the logic that adds the prefix (and suffix, depending on the second parameter) into `validateCode()` rather than changing every call site.